dashboard/src/actions/repos.ts

```diff
@@ -359,13 +359,18 @@ export function findPackageInRepo(
 ): ThunkAction<Promise<boolean>, IStoreState, null, AppReposAction> {
   return async dispatch => {
     dispatch(requestRepo());
-    if (app?.availablePackageRef?.context?.namespace && app?.availablePackageRef?.identifier) {
+    if (
+      app?.availablePackageRef?.context?.namespace &&
+      app?.availablePackageRef?.identifier &&
+      app?.availablePackageRef?.plugin
```

Comment:
Also, not sure if I'm reading the indentation correctly, but looks like we dispatch an error saying that the chart could not be found, rather than that not enough info was provided below, for the `else` of this condition?

Reply:
wrt the first comment, yep, fixed. Once we use the `...packageRef` we no longer need this check (which was mainly required in typescript, given that some field could eventually be undefined - although we could have just appended `!` at the end to force the type assertion-)

This function is used to, given an InstalledPackage, trace it back and retrieve the proper details. However, take into account that, we are here (in this function) because we don't have enough data to populate the upgrade form (usually pkg values, pkg id or pkg version).

Even if a manual repo disambiguation should be enough, we are not always on the ideal case. Let's see:

1. Ideal scenario: an app has been installed from a repo that still exists, still contains the package and it has been installed with a plugin, but we have to disambiguate manually. 
2. Bad, but recoverable, scenario: the repo doesn't exist or it no longer contains this app. Once the user adds a new repo containing the item, everything backs to normal.
3. Worst case, unknown package: there is an unknown package (aka, w/o any availablePackage reference or plugin). In that case, just show "ey, this app cannot be upgraded"

However, the underlying problem was that we were actually allowing those weird cases. I mean, why would the UI allow  upgrading something we know is not upgradeable whatsoever. So, in this PR (I can do it separately if needed), I'm disabling the upgrade/rollback buttons and rendering the AppView, so that users don't see "Error: xxx", but a handy warning.

Demo here:


https://user-images.githubusercontent.com/11535726/133310239-1f6d4798-24b7-41ac-b3d0-86840cf660f8.mp4

Reply:
Thanks for the demo.

>     3\. Worst case, unknown package: there is an unknown package (aka, w/o any availablePackage reference or plugin).

I'm still unsure why we'd ever be missing a plugin here (regardless of whether the page reloads). We fetch the installed package via a plugin - it should be an error on our side if there's no plugin. But yes, if the plugin returns the installed package without an available package reference, then it was unable to find one.

> In that case, just show "ey, this app cannot be upgraded"

Rather than "There is a problem with this package: this package has missing information, some actions might not be available", perhaps we can give a way forward... something like: "Kubeapps was unable to find the source from which "kubo" was installed. You may need to re-add the source to enable some actions." Hmm, but it is tricky keeping that general.


> 
> 
> However, the underlying problem was that we were actually allowing those weird cases. I mean, why would the UI allow upgrading something we know is not upgradeable whatsoever.

Yeah, similar to displaying a selection of releases which may not even exist to roll-back to.

> So, in this PR (I can do it separately if needed), I'm disabling the upgrade/rollback buttons and rendering the AppView, so that users don't see "Error: xxx", but a handy warning.

Great.